src/nni_manager/training_service/pai/paiTrainingService.ts

```diff
@@ -213,86 +192,6 @@ class PAITrainingService implements TrainingService {
-                    port: 80,
-                    path: '/webhdfs/api/v1',
-                    host: this.paiClusterConfig.host
-                });
-                if(this.paiClusterConfig.passWord) {
-                    // Get PAI authentication token
-                    await this.updatePaiToken();
-                } else if(this.paiClusterConfig.token) {
-                    this.paiToken = this.paiClusterConfig.token;
-                } else {
-                    deferred.reject(new Error('pai cluster config format error, please set password or token!'));
-                }
-
-                deferred.resolve();
-                break;
-
-            case TrialConfigMetadataKey.TRIAL_CONFIG:
-                if (this.paiClusterConfig === undefined) {
-                    this.log.error('pai cluster config is not initialized');
-                    deferred.reject(new Error('pai cluster config is not initialized'));
-                    break;
-                }
-                this.paiTrialConfig = <NNIPAITrialConfig>JSON.parse(value);
-
-                // Validate to make sure codeDir doesn't have too many files
-                try {
-                    await validateCodeDir(this.paiTrialConfig.codeDir);
-                } catch (error) {
-                    this.log.error(error);
-                    deferred.reject(new Error(error));
-                    break;
-                }
-           
-                // Copy experiment files from local folder to HDFS
-                this.copyExpCodeDirPromise = HDFSClientUtility.copyDirectoryToHdfs(
-                    this.paiTrialConfig.codeDir,
-                    HDFSClientUtility.getHdfsExpCodeDir(this.paiClusterConfig.userName),
-                    this.hdfsClient
-                );
-                
-                // Upload authFile to hdfs
-                if (this.paiTrialConfig.authFile) {
-                    this.authFileHdfsPath = unixPathJoin(HDFSClientUtility.hdfsExpRootDir(this.paiClusterConfig.userName), 'authFile');
-                    this.copyAuthFilePromise = HDFSClientUtility.copyFileToHdfs(this.paiTrialConfig.authFile, this.authFileHdfsPath, this.hdfsClient);
-                }
-
-                deferred.resolve();
-                break;
-            case TrialConfigMetadataKey.VERSION_CHECK:
-                this.versionCheck = (value === 'true' || value === 'True');
-                break;
-            case TrialConfigMetadataKey.LOG_COLLECTION:
-                this.logCollection = value;
-                break;
-            case TrialConfigMetadataKey.MULTI_PHASE:
-                this.isMultiPhase = (value === 'true' || value === 'True');
-                break;
-            default:
-                //Reject for unknown keys
-                throw new Error(`Uknown key: ${key}`);
-        }
-
-        return deferred.promise;
-    }
-
     public getClusterMetadata(key: string): Promise<string> {
```

Comment:
It seems this getClusterMetadata is not implemented, then throw not implemented error.

Reply:
fixed.